### PR TITLE
Fix Vsync Via Compositor causing halved refresh rate when vsync is forced by the graphics driver

### LINKED ISSUE #35038: 3.2 beta 6: enabling both use_vsync and sync_via_compositor halves the refresh rate of windowed games
**Godot version:**
3.2.beta6

**OS/device including version:**
Windows 10 1909
Radeon VII
Adrenalin 19.9.3, 20.1.1

**Issue description:**
Enabling both _use_vsync_ and _sync_via_compositor_ halves the refresh rate of windowed games. Also occurs in the editor. Issue may be present in earlier builds.

### PR DESCRIPTION
**Pull request by hoontee**
Reverts commit 52f44e72b61a8e9e923454bacdcf62de1b1c6309.

Add check to prevent compositor sync if the graphics driver is forcing vsync.

Fixes #35038.

(Addendum: this PR does not negatively impact users unaffected by #35038.)

### CODE REVIEW

**Comment by akien-mga** [comment]
CC @TerminalJack 

@Jakob-PB Would you be able to test this PR to confirm that it also fixes your issue from #35065?

**Comment by Jakob-PB** [comment]
First time building the engine for me (it was impressively easy for a C++ project I must say 😄). I downloaded the source up to this commit from [here](https://github.com/godotengine/godot/tree/59c2f42f6cd7d639c163d60473d026162a6bae87), as the zip option and compiled for windows with SCons. Let me know if something about that is wrong, but it seems like a valid option.

Anyway, assuming I compiled the right thing this PR does not fix the issue on my end. I'm seeing the same frame rate halving behavior in the editor while `Vsync Via Compositor` is on.

**Comment by hoontee** [comment]
That's a shame. I trust you compiled it correctly so I'm closing this PR and going back to find something that will fix the halved framerate for both AMD and Nvidia cards.

**Comment by hoontee** [comment]
Reopened. See https://github.com/godotengine/godot/issues/35065#issuecomment-574451463

**Comment by hoontee** [comment]
CC @TerminalJack

**Comment by akien-mga** [comment]
Given #35065, I suggest to keep it disabled by default for now, but the extra fix for #35038 seems good.

**Comment by TerminalJack** [comment]
I compiled and ran the PR without any issues.  The vsync_via_compositor setting still works when enabled.

I cannot confirm that it will prevent the halving of the framerate when the driver is set to force vsync, however.  The Nvidia Control Panel on the system that is affected by the stutter issue doesn't work (see the rating and reviews [here](https://www.microsoft.com/en-us/p/nvidia-control-panel/9nf8h0h7wmlt?rtc=1&&activetab=pivot:reviewstab)) so I have no way to change any of the driver settings.

Basically, Nvidia has somehow managed to completely FUBAR the Control Panel by requiring it to be downloaded from the Windows Store.  It also probably doesn't help that I have a Microsoft-specific custom build of the drivers since the system in-question is a Microsoft Surface Studio 2.

**Comment by hoontee** [comment]
@akien-mga I don't know why my changes aren't showing up so I'm just gonna start a new PR.
